feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java

```diff
@@ -20,165 +34,196 @@
+		((HttpRequestBase) request).setURI(uri);
+		logger.info("URI :{}",uri);
+		request.addHeader("Content-Type", content_type);
+		request.addHeader("Accept", accept);
+		request.addHeader("company", company);
+		request.addHeader("Authorization", "Bearer " + auth);
+		response = httpClient.execute(request);
+		logger.info("Status Code  : {}", response.getStatusLine().getStatusCode());
+		response.close();
+		httpClient.close();
+
+	}
+
+	public void update_customer(OpenMRSPatientFullRepresentation patientFR, String auth) throws Exception {
+		String name = patientFR.getPerson().getPreferredName().getPreferredFullName();
+		String uuid = getuuid(patientFR);
+	
+		List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>();
+		nameValuePairs.add(new BasicNameValuePair("name", name));
+		nameValuePairs.add(new BasicNameValuePair("contact_name", uuid));
+		nameValuePairs.add(new BasicNameValuePair("currency_id", currencyId));
+		
+		OpenMRSPersonAddress address = patientFR.getPerson().getPreferredAddress();
+		nameValuePairs.add(new BasicNameValuePair("billing[name]", address.getAddress1()));
+		nameValuePairs.add(new BasicNameValuePair("billing[address_street_1]", address.getAddress2()));
+		nameValuePairs.add(new BasicNameValuePair("billing[address_street_2]", address.getAddress3()));
+		nameValuePairs.add(new BasicNameValuePair("billing[city]", address.getCityVillage()));
+		nameValuePairs.add(new BasicNameValuePair("billing[state]", address.getStateProvince()));
+		nameValuePairs.add(new BasicNameValuePair("billing[country_id]", address.getCountry()));
+		nameValuePairs.add(new BasicNameValuePair("billing[zip]", address.getPostalCode().toString()));
+		
+		List<OpenMRSPersonAttribute> attribute=patientFR.getPerson().getAttributes();
+		for (OpenMRSPersonAttribute openMRSPersonAttribute : attribute) {
+			String display=openMRSPersonAttribute.getAttributeType().getDisplay();
+			String value = openMRSPersonAttribute.getValue().toString();
+			if(display.equals("phoneNumber")) {
+			nameValuePairs.add(new BasicNameValuePair("phone",value));
+			}
+			else
+			nameValuePairs.add(new BasicNameValuePair(display,value));
+		}
+		String id = this.get_list_customers(uuid, auth);
+		
+		CloseableHttpResponse response;
+		CloseableHttpClient httpClient = HttpClients.createDefault();
+		HttpPut request = new HttpPut(crater_url + "/api/v1/customers/"+id);	
+		
+		
+		URI uri = new URIBuilder(request.getURI()).setParameters(nameValuePairs).build();
+		((HttpRequestBase) request).setURI(uri);
+		logger.info("URI :{}",uri);
+		request.addHeader("Content-Type", content_type);
+		request.addHeader("Accept", accept);
+		request.addHeader("company", company);
+		request.addHeader("Authorization", "Bearer " + auth);
+		response = httpClient.execute(request);
+		logger.info("Status Code  : {}", response.getStatusLine().getStatusCode());
+		response.close();
+		httpClient.close();
```

Comment:
This code is also duplicate and can be extracted into sendRequestToCrate( ) method, and reused across both create() and update() methods, by passing the nameValuePairs as params.